### PR TITLE
fix(core): Adds express/fastify to peer dependency list and makes them optional

### PR DESCRIPTION
**Pull request by kamronbatman**
Adds express/fastify to the peer dependency list to fix a problem with yarn pnp workspaces and resolving dependencies from @nestjs/common through @nestjs/serve-static.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features) **(Not Applicable)**
- [ ] Docs have been added / updated (for bug fixes / features) **(Not Applicable)**

## PR Type
fix: Fixes a Nestjs runtime error saying it cannot find express when using yarn pnp workspaces.

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## What is the current behavior?
Receive the error: "The "express" package is missing. Please, make sure to install this library ($ npm install express) to take advantage of ServeStaticModule."

Screen Shot 2022-10-07 at 1 53 13 PM

Issue Number: N/A

## What is the new behavior?
The `ServeStaticModule` loads properly and this error does not show up.

## Does this PR introduce a breaking change?
- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by kamronbatman** [comment]
@kamilmysliwiec, since this is my first PR, I am not sure if I filled it out correctly. Let me know if I missed anything.

File: package.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
   },
   "peerDependencies": {
     "@nestjs/common": "^9.0.0",
-    "@nestjs/core": "^9.0.0"
+    "@nestjs/core": "^9.0.0",
+    "express": "^4.8.1"
```
Comment (micalevisk) [inline package.json:23]: `express` is optional here. You could add `fastify` as well and mark both of them as optional with https://docs.npmjs.com/cli/v8/configuring-npm/package-json#peerdependenciesmeta

Comment (kamronbatman) [reply package.json:23]: Great idea. I'll do that and test it out with yarn pnp workspaces.

Comment (kamronbatman) [reply package.json]: @micalevisk I updated with the following:
* I had the wrong express version, so fixed that.
* Added fastify and fastify/startic.
* Added all 3 to peerDependenciesMeta so they are optional

**Comment by kamilmysliwiec** [comment]
lgtm

**Pull request merged**